src/Workbench.ts

```diff
@@ -262,11 +282,32 @@ export default class Workbench extends ThemedBase<WorkbenchProperties> {
 	protected onAttach() {
 		window.addEventListener('resize', this._onresize);
 		window.addEventListener('beforeunload', this._onbeforeunload);
+		this._emitErrorsSub = project.emitErrors.subscribe((errors: EmitError[]) => {
+			errors.map((error) => {
+				this._addConsoleMessage({
+					type: ConsoleMessageType.Error,
+					message: error.message,
+					lineNumber: error.lineNumber,
+					filename: error.filename
+				});
+			});
+		});
 	}
 
 	protected onDetach() {
 		window.removeEventListener('resize', this._onresize);
 		window.removeEventListener('beforeunload', this._onbeforeunload);
+		this._emitErrorsSub && this._emitErrorsSub.unsubscribe();
+	}
+
+	private _addConsoleMessage(message: ConsoleMessage) {
+		this._messages = [ ...this._messages, message ];
```

Comment:
I love ES6 syntax, but it seems excessive here... especially with the performance implications of generating multiple arrays and iterating over a source array.  What is wrong with good ole `this._messages.push()`?

Reply:
I am kind of baffled by this, but it doesn't work if I change the line to `this._messages.push(message);`. Debugging, I can see that the array is getting populated, but the UI never actually renders the log statements. Is there something I'm not understanding about maquette here?
![devtools](https://user-images.githubusercontent.com/293805/33640834-8095115e-d9f8-11e7-91dc-2d9c01c2ee99.png)

Reply:
So it seems like the diff has issues because the `oldValue` and `newValue` args being passed to the diff function are the exact same arrays, with the exact same arrays. For this reason, it looks like I do need to set it to a new array.

To test this, I created a custom diff function and set it with the `@diffProperty` decorator.

```ts
function diffMessages(oldValue: ConsoleMessage[] = [], newValue: ConsoleMessage[] = []) {
        // this always logs true
	console.log(oldValue === newValue && oldValue.length === newValue.length); true
        return { ... };
}
```

Reply:
As discussed offline, I take back my comment because of the way properties are diffed... iterating over the array to determine if something has changed in a custom diff is just as _bad_ creating a new array every time it changes.

Reply:
Ok, cool. Thanks for the feedback! It was good to learn more about the diffing strategies!